From 00403bca157d521e5d34509d193e6f59390421bf Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 6 Mar 2021 22:13:29 +0000
Subject: [PATCH] gnu: Add re

* gnu/package/lisp-xyz.scm: (sbcl-re, cl-re, ecl-re): New variables
---
 gnu/packages/lisp-xyz.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 863925a4c0..dafbe65147 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -609,6 +609,37 @@ compatible with ANSI-compliant Common Lisp implementations.")
 (define-public cl-parse
   (sbcl-package->cl-source-package sbcl-parse))
 
+(define-public sbcl-re
+  (let ((commit "cfbc1f482970221e80d445080a188fd5c755cd2c")
+        (revision "1"))
+    (package
+     (name "sbcl-re")
+      (version (git-version "1.0.0" revision commit))
+     (source
+      (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/massung/re")
+             (commit commit)))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1y2gq2sckspnq8118bix55p2j43dk9qn3p8a2rplp1ip2qxqbb1i"))))
+     (build-system asdf-build-system/sbcl)
+     (inputs
+      `(("parse" ,sbcl-parse)))
+     (home-page "https://github.com/massung/re")
+     (synopsis "Lua-style Pattern Matching for Common Lisp")
+     (description
+      "RE is a small, portable, lightweight, and quick, regular
+expression library for Common Lisp.  It is a non-recursive, backtracing VM.")
+     (license license:asl2.0))))
+
+(define-public ecl-re
+  (sbcl-package->ecl-package sbcl-re))
+
+(define-public cl-re
+  (sbcl-package->cl-source-package sbcl-re))
+
 (define-public sbcl-ubiquitous
   (let ((commit "35eb7bd9e1b3daee1705f6b41260775180cce8af")
         (revision "1"))
-- 
2.29.2

